src/solvers/smt2/smt2_parser.cpp

```diff
@@ -965,18 +965,21 @@ typet smt2_parsert::sort()
   }
 }
 
-typet smt2_parsert::function_signature_definition()
+std::pair<typet, std::vector<irep_idt>>
```

Comment:
I think this would deserve a class or struct of its own. `first` and `second` are not very meaningful here. With such a class or struct it would also be easy to add consistency constraints.